src/github.com/matrix-org/dendrite/federationsender/storage/room_table.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2017 Vector Creations Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"database/sql"
+)
+
+const roomSchema = `
+CREATE TABLE IF NOT EXISTS rooms (
+    -- The string ID of the room
+    room_id TEXT NOT NULL CONSTRAINT room_id_unique UNIQUE,
+    -- The most recent event state by the room server.
+    -- We can use this to tell if our view of the room state has become
+    -- desynchronised.
+    last_event_id TEXT NOT NULL
+);`
+
+const insertRoomSQL = "" +
+	"INSERT INTO rooms (room_id, last_event_id)" +
+	" VALUES ($1, '')" +
+	" ON CONFLICT ON CONSTRAINT room_id_unique" +
+	" DO NOTHING"
```

Comment:
Because I want to use an explicit "SELECT FOR UPDATE" to get a row lock.
(I think the INSERT will take get a row lock anyway, but I would prefer to be explicit)